zerver/tests/test_channel_creation.py

```diff
@@ -266,6 +269,309 @@ def test_add_stream_as_default_on_stream_creation(self) -> None:
         )
         self.assert_json_error(result, "A default channel cannot be private.")
 
+    def check_created_channel_exists_with_property(
```

Comment:
I don't think this helper is needed. You just need to chec properties when creating stream with message retention days value or when creating a default stream, or when setting a color. And it is just two lines of code, a `get_stream` statement and a assertEqual check.